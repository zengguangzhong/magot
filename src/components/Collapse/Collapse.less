@import '../../variable.less';

.@{prefix}-collapse {
  position: relative;
  background-color: @collapse-background-color;
  border: 1px solid @collapse-border-color;
  border-radius: @border-radius-base;
  .box();
  .box-sizing();
}

.@{prefix}-collapse-panel {
  border-bottom: 1px solid @collapse-border-color;

  &:last-child {
    border-bottom: none;
  }

  &[open] {
    .@{prefix}-collapse-panel-header {
      & > .@{prefix}-icon {
        transform: rotate(0);
      }
    }
  }
}

.@{prefix}-collapse-panel-disabled {
  pointer-events: none;
  cursor: not-allowed;
  .@{prefix}-collapse-panel-header {
    color: @collapse-header-color-disabled;
  }
}

.@{prefix}-collapse-panel-header {
  padding: @collapse-header-padding;
  font-size: @collapse-header-font-size;
  color: @collapse-header-color;
  outline: none;
  cursor: pointer;

  &::-webkit-details-marker {
    display: none;
  }

  & > .@{prefix}-icon {
    margin-right: @collapse-header-title-gap;
    transform: rotate(-90deg);
    .transition(transform);
  }

  & > .title {
    display: inline-block;
    font-weight: 400;
    .box();
  }
}

.@{prefix}-collapse-panel-content {
  padding: @collapse-content-padding;
  background-color: @collapse-content-background-color;
  border-top: 1px solid @collapse-border-color;
  overflow: hidden;
  .transition();
}

.@{prefix}-collapse-borderless {
  border-left: none;
  border-top: none;
  border-right: none;
  border-radius: 0;
  background-color: transparent;
  .@{prefix}-collapse-panel-content {
    padding-top: 0;
    border-top: none;
  }
}
