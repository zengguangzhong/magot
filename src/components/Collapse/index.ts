export { default, CollapseProps, CollapsePanelProps } from './Collapse';
