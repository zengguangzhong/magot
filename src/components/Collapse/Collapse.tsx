import React from 'react';

import Icon from '../Icon';
import * as component from '../component';

import './Collapse.less';

export interface CollapseProps
  extends component.BaseComponent,
    component.NestedComponent {
  /**
   * 激活展开的面板，当是手风琴模式时，只能展开一个面板
   */
  active?: number | number[];

  /**
   * 是否是手风琴模式的折叠面板，同时只能展开一个面板
   * @default false
   */
  accordion?: boolean;

  /**
   * 带边框的折叠面板
   * @default true
   */
  bordered?: boolean;

  /**
   * 折叠面板列表
   */
  children?: React.FunctionComponentElement<CollapsePanelProps>[];

  /**
   * 切换面板后的回调函数
   */
  onChange?: (panels: number[]) => void;
}

export interface CollapsePanelProps
  extends component.BaseComponent,
    component.DisableComponent,
    component.NestedComponent {
  /**
   * 面板标题
   */
  title: string;

  /**
   * 是否显示面板上的箭头图标
   */
  showArrow?: boolean;

  /**
   * 是否展开该面板
   */
  open?: boolean;

  /**
   * 面板展开事件回调函数
   */
  onOpen?: () => void;

  /**
   * 面板收起事件回调函数
   */
  onClose?: () => void;
}

const defaultProps: Partial<CollapseProps> = {
  bordered: true,
  accordion: false,
};

const defaultPanelProps: Partial<CollapsePanelProps> = {
  ...component.getDefaultDisabledProps(),
  showArrow: true,
};

function CollapsePanel(props: CollapsePanelProps) {
  const handleToggle = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    const detailsNode = e.currentTarget.parentNode;
    if (!detailsNode) return;
    const open = (detailsNode as HTMLDetailsElement).open;
    const cb = open ? props.onClose : props.onOpen;
    cb && cb();
  };

  const type = 'collapse-panel';
  const prefix = component.getComponentPrefix(type);
  const cls = component.getComponentClasses(type, props);

  return (
    <details className={cls} style={props.style} open={props.open}>
      <summary className={prefix + '-header'} onClick={handleToggle}>
        {props.showArrow && <Icon name="caret" />}
        <h4 className="title">{props.title}</h4>
      </summary>
      <section className={prefix + '-content'}>{props.children}</section>
    </details>
  );
}

CollapsePanel.defaultProps = defaultPanelProps;

function Collapse(props: CollapseProps) {
  const children = React.Children.toArray(props.children || []);

  const active = props.active;
  let activeds =
    active !== void 0 ? (Array.isArray(active) ? active : [active]) : [];
  activeds = activeds.filter(i => !children[i].props.disabled);

  const [activedPanels, setActivedPanels] = React.useState(activeds);

  const changeActivedPanels = (panels: number[]) => {
    setActivedPanels(panels);
    props.onChange && props.onChange(panels);
  };

  const handlePanelOpen = (index: number) => {
    const panels = props.accordion ? [index] : [...activedPanels, index];
    changeActivedPanels(panels);
  };

  const handlePanelClose = (index: number) => {
    if (props.accordion) {
      changeActivedPanels([]);
      return;
    }
    const panels = activedPanels.filter(p => p !== index);
    changeActivedPanels(panels);
  };

  let panels = children.map((panel, index) => {
    return React.cloneElement(panel, {
      open: activedPanels.includes(index),
      onOpen: () => {
        handlePanelOpen(index);
        panel.props.onOpen && panel.props.onOpen();
      },
      onClose: () => {
        handlePanelClose(index);
        panel.props.onClose && panel.props.onClose();
      },
    });
  });

  const prefix = component.getComponentClasses('collapse');
  const cls = component.getComponentClasses('collapse', props, {
    [prefix + '-borderless']: !props.bordered,
  });

  return (
    <div className={cls} style={props.style}>
      {panels}
    </div>
  );
}

Collapse.defaultProps = defaultProps;
Collapse.Panel = CollapsePanel;

export default Collapse;
