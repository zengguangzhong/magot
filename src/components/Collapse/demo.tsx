import React from 'react';
import Collapse from './Collapse';
import { Link } from 'react-router-dom';

function CollapseDemo() {
  const text =
    'This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.';
  const textStyle = { color: 'rgba(0,0,0,.65)', fontSize: 14, margin: 0 };
  return (
    <>
      <Link to="/" className="demo-goback">
        返回
      </Link>
      <div className="demo-box">
        <Collapse active={[0, 1]} onChange={console.log}>
          <Collapse.Panel title="Panel Title1">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title2">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title3" disabled={true}>
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
        </Collapse>
      </div>
      <div className="demo-box">
        <Collapse accordion={true} active={0} onChange={console.log}>
          <Collapse.Panel title="Panel Title1">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title2">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title3">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
        </Collapse>
      </div>
      <div className="demo-box">
        <Collapse bordered={false} onChange={console.log}>
          <Collapse.Panel title="Panel Title1">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title2">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title3">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
        </Collapse>
      </div>
      <div className="demo-box">
        <Collapse onChange={console.log}>
          <Collapse.Panel title="Panel Title1">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title2" showArrow={false}>
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
        </Collapse>
      </div>
      <div className="demo-box">
        <Collapse active={0} onChange={console.log}>
          <Collapse.Panel title="Panel Title1">
            <Collapse active={0} onChange={console.log}>
              <Collapse.Panel title="SubPanel Title1">
                <p style={textStyle}>{text}</p>
              </Collapse.Panel>
              <Collapse.Panel title="SubPanel Title2">
                <p style={textStyle}>{text}</p>
              </Collapse.Panel>
            </Collapse>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title2">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
          <Collapse.Panel title="Panel Title3">
            <p style={textStyle}>{text}</p>
          </Collapse.Panel>
        </Collapse>
      </div>
    </>
  );
}

export default CollapseDemo;
