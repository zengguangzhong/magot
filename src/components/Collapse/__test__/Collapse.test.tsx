import React from 'react';
import { render, mount } from 'enzyme';
import Collapse from '../Collapse';
import { injectStyleTestSuites } from '../../../../test/shared.test';

describe('Collapse', () => {
  const text =
    'This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.';

  it('should renders dom correctly', () => {
    const wrapper = render(<Collapse />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders have children', () => {
    const wrapper = render(
      <Collapse>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3" disabled={true}>
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-collapse-panel')).toHaveLength(3);
  });

  it('should renders `active` when pass in', () => {
    const wrapper = render(
      <Collapse active={[0, 1]}>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3">
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.mgt-collapse-panel[open]')).toHaveLength(2);
  });

  it('should renders `accordion` is true', () => {
    const wrapper = mount(
      <Collapse accordion={true} active={0}>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3">
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    const panels = wrapper.find('.mgt-collapse-panel');
    const first = panels.first();
    const last = panels.last();
    expect(wrapper).toMatchSnapshot();
    expect(first.render().prop('open')).toEqual(true);
    last.find('summary').simulate('click');
    expect(last.render().prop('open')).toEqual(true);
  });

  it('should renders `bordered` is false', () => {
    const wrapper = render(
      <Collapse bordered={false}>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3">
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('mgt-collapse-borderless')).toBe(true);
  });

  it('should simulates change event', () => {
    const handleChange = jest.fn();
    const wrapper = mount(
      <Collapse active={[0]} onChange={handleChange}>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3">
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    wrapper
      .find('.mgt-collapse-panel')
      .at(2)
      .find('summary')
      .simulate('click');
    expect(handleChange).toBeCalledWith([0, 2]);
  });

  it('should renders `showArrow` is false', () => {
    const wrapper = render(
      <Collapse>
        <Collapse.Panel title="Panel Title1">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2" showArrow={false}>
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find('.mgt-collapse-panel')
        .first()
        .find('.mgt-icon')
    ).toHaveLength(1);
    expect(
      wrapper
        .find('.mgt-collapse-panel')
        .last()
        .find('.mgt-icon')
    ).toHaveLength(0);
  });

  it('should renders nested Collapses', () => {
    const wrapper = render(
      <Collapse active={0}>
        <Collapse.Panel title="Panel Title1">
          <Collapse active={0}>
            <Collapse.Panel title="SubPanel Title1">
              <p>{text}</p>
            </Collapse.Panel>
            <Collapse.Panel title="SubPanel Title2">
              <p>{text}</p>
            </Collapse.Panel>
          </Collapse>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title2">
          <p>{text}</p>
        </Collapse.Panel>
        <Collapse.Panel title="Panel Title3">
          <p>{text}</p>
        </Collapse.Panel>
      </Collapse>
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find('>.mgt-collapse-panel')
        .first()
        .find('.mgt-collapse')
    ).toHaveLength(1);
  });

  injectStyleTestSuites(Collapse);
});
