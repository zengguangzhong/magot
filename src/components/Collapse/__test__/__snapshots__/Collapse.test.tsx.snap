// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should renders \`accordion\` is true 1`] = `
<Collapse
  accordion={true}
  active={0}
  bordered={true}
>
  <div
    className="mgt-collapse"
  >
    <CollapsePanel
      disabled={false}
      key=".0"
      onClose={[Function]}
      onOpen={[Function]}
      open={true}
      showArrow={true}
      title="Panel Title1"
    >
      <details
        className="mgt-collapse-panel"
        open={true}
      >
        <summary
          className="mgt-collapse-panel-header"
          onClick={[Function]}
        >
          <Icon
            name="caret"
            spin={false}
          >
            <i
              className="mgt-icon mgt-icon-caret"
              style={Object {}}
            />
          </Icon>
          <h4
            className="title"
          >
            Panel Title1
          </h4>
        </summary>
        <section
          className="mgt-collapse-panel-content"
        >
          <p>
            This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
          </p>
        </section>
      </details>
    </CollapsePanel>
    <CollapsePanel
      disabled={false}
      key=".1"
      onClose={[Function]}
      onOpen={[Function]}
      open={false}
      showArrow={true}
      title="Panel Title2"
    >
      <details
        className="mgt-collapse-panel"
        open={false}
      >
        <summary
          className="mgt-collapse-panel-header"
          onClick={[Function]}
        >
          <Icon
            name="caret"
            spin={false}
          >
            <i
              className="mgt-icon mgt-icon-caret"
              style={Object {}}
            />
          </Icon>
          <h4
            className="title"
          >
            Panel Title2
          </h4>
        </summary>
        <section
          className="mgt-collapse-panel-content"
        >
          <p>
            This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
          </p>
        </section>
      </details>
    </CollapsePanel>
    <CollapsePanel
      disabled={false}
      key=".2"
      onClose={[Function]}
      onOpen={[Function]}
      open={false}
      showArrow={true}
      title="Panel Title3"
    >
      <details
        className="mgt-collapse-panel"
        open={false}
      >
        <summary
          className="mgt-collapse-panel-header"
          onClick={[Function]}
        >
          <Icon
            name="caret"
            spin={false}
          >
            <i
              className="mgt-icon mgt-icon-caret"
              style={Object {}}
            />
          </Icon>
          <h4
            className="title"
          >
            Panel Title3
          </h4>
        </summary>
        <section
          className="mgt-collapse-panel-content"
        >
          <p>
            This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
          </p>
        </section>
      </details>
    </CollapsePanel>
  </div>
</Collapse>
`;

exports[`Collapse should renders \`active\` when pass in 1`] = `
<div
  class="mgt-collapse"
>
  <details
    class="mgt-collapse-panel"
    open=""
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title1
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
    open=""
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title2
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title3
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
</div>
`;

exports[`Collapse should renders \`bordered\` is false 1`] = `
<div
  class="mgt-collapse mgt-collapse-borderless"
>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title1
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title2
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title3
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
</div>
`;

exports[`Collapse should renders \`showArrow\` is false 1`] = `
<div
  class="mgt-collapse"
>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title1
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <h4
        class="title"
      >
        Panel Title2
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
</div>
`;

exports[`Collapse should renders custom className 1`] = `
<div
  class="mgt-collapse jest-test-class"
/>
`;

exports[`Collapse should renders custom styles 1`] = `
<div
  class="mgt-collapse"
  style="color:#333;background-color:#f0f0f0"
/>
`;

exports[`Collapse should renders dom correctly 1`] = `
<div
  class="mgt-collapse"
/>
`;

exports[`Collapse should renders have children 1`] = `
<div
  class="mgt-collapse"
>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title1
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title2
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel mgt-collapse-panel-disabled"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title3
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
</div>
`;

exports[`Collapse should renders nested Collapses 1`] = `
<div
  class="mgt-collapse"
>
  <details
    class="mgt-collapse-panel"
    open=""
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title1
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <div
        class="mgt-collapse"
      >
        <details
          class="mgt-collapse-panel"
          open=""
        >
          <summary
            class="mgt-collapse-panel-header"
          >
            <i
              class="mgt-icon mgt-icon-caret"
            />
            <h4
              class="title"
            >
              SubPanel Title1
            </h4>
          </summary>
          <section
            class="mgt-collapse-panel-content"
          >
            <p>
              This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
            </p>
          </section>
        </details>
        <details
          class="mgt-collapse-panel"
        >
          <summary
            class="mgt-collapse-panel-header"
          >
            <i
              class="mgt-icon mgt-icon-caret"
            />
            <h4
              class="title"
            >
              SubPanel Title2
            </h4>
          </summary>
          <section
            class="mgt-collapse-panel-content"
          >
            <p>
              This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
            </p>
          </section>
        </details>
      </div>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title2
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
  <details
    class="mgt-collapse-panel"
  >
    <summary
      class="mgt-collapse-panel-header"
    >
      <i
        class="mgt-icon mgt-icon-caret"
      />
      <h4
        class="title"
      >
        Panel Title3
      </h4>
    </summary>
    <section
      class="mgt-collapse-panel-content"
    >
      <p>
        This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content, This is panel content.
      </p>
    </section>
  </details>
</div>
`;
